pkg/cloud_provider/file/file.go

```diff
@@ -260,55 +260,7 @@ func NewGCFSService(version string, client *http.Client, primaryFilestoreService
 }
 
 func (manager *gcfsServiceManager) CreateInstance(ctx context.Context, obj *ServiceInstance) (*ServiceInstance, error) {
-	betaObj := &filev1beta1.Instance{
-		Tier: obj.Tier,
-		FileShares: []*filev1beta1.FileShareConfig{
-			{
-				Name:             obj.Volume.Name,
-				CapacityGb:       util.RoundBytesToGb(obj.Volume.SizeBytes),
-				NfsExportOptions: extractNfsShareExportOptions(obj.NfsExportOptions),
-			},
-		},
-		Networks: []*filev1beta1.NetworkConfig{
-			{
-				Network:         obj.Network.Name,
-				Modes:           []string{"MODE_IPV4"},
-				ReservedIpRange: obj.Network.ReservedIpRange,
-				ConnectMode:     obj.Network.ConnectMode,
-			},
-		},
-		KmsKeyName: obj.KmsKeyName,
-		Labels:     obj.Labels,
-	}
-
-	klog.V(4).Infof("Creating instance %q: location %q, tier %q, capacity %v, network %q, ipRange %q, connectMode %q, KmsKeyName %q, labels %v",
-		obj.Name,
-		obj.Location,
-		betaObj.Tier,
-		betaObj.FileShares[0].CapacityGb,
-		betaObj.Networks[0].Network,
-		betaObj.Networks[0].ReservedIpRange,
-		betaObj.Networks[0].ConnectMode,
-		betaObj.KmsKeyName,
-		betaObj.Labels)
-	op, err := manager.instancesService.Create(locationURI(obj.Project, obj.Location), betaObj).InstanceId(obj.Name).Context(ctx).Do()
-	if err != nil {
-		klog.Errorf("CreateInstance operation failed for instance %s: %w", obj.Name, err)
-		return nil, err
-	}
-
-	klog.V(4).Infof("For instance %v, waiting for create instance op %v to complete", obj.Name, op.Name)
-	err = manager.waitForOp(ctx, op)
-	if err != nil {
-		klog.Errorf("WaitFor CreateInstance op %s failed: %w", op.Name, err)
-		return nil, err
-	}
-	instance, err := manager.GetInstance(ctx, obj)
-	if err != nil {
-		klog.Errorf("failed to get instance after creation: %w", err)
-		return nil, err
-	}
-	return instance, nil
+	return manager.CreateInstanceFromBackupSource(ctx, obj, "")
```

Comment:
the name is confusing

Reply:
Converged into one CreateInstance method